File: apps/cargo-tracker/src/main/java/com/example/cargotracker/tracking/domain/repository/TrackingRepository.java
package com.example.cargotracker.tracking.domain.repository;

import com.example.cargotracker.tracking.domain.model.aggregates.TrackingEntry;
import com.example.cargotracker.tracking.domain.model.valueobjects.TrackingNumber;

import java.util.Optional;
import java.util.UUID;

public interface TrackingRepository {
    void save(TrackingEntry entry);
    Optional<TrackingEntry> findByTrackingNumber(TrackingNumber trackingNumber);
    Optional<TrackingEntry> findByBookingId(UUID bookingId);
}

File: apps/cargo-tracker/packs/tracking/app/domain/tracking/domain/tracking_repository.rb
# frozen_string_literal: true

module Tracking
  module Domain
    # 追跡活動リポジトリの抽象ポート（出力ポート・ADR-0001）。
    class TrackingRepository
      def save(_activity) = raise NotImplementedError
      def find_by_booking_id(_booking_id) = raise NotImplementedError
      def find_by_tracking_number(_tracking_number) = raise NotImplementedError
      def exists_for_booking?(_booking_id) = raise NotImplementedError
    end
  end
end
